src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -1641,7 +1641,17 @@ def process_disk_or_snapshot_create_namespace(cmd, namespace):
                 if not source_info:
                     from azure.cli.core.util import parse_proxy_resource_id
                     result = parse_proxy_resource_id(namespace.source_disk or namespace.source_snapshot)
-                    source_info, _ = _get_disk_or_snapshot_info(cmd.cli_ctx, result['resource_group'], result['name'])
+                    try:
+                        source_info, _ = _get_disk_or_snapshot_info(cmd.cli_ctx,
+                                                                    result['resource_group'],
+                                                                    result['name'])
+                    except ResourceNotFoundError:
```

Comment:
Should we skip autodetection not only for `ResourceNotFoundError` but also other exceptions? @zhoxing-ms any suggestions?

Reply:
May I ask if there are other exceptions, can the subsequent process still do `copy_start` normally?

Reply:
If there are other exceptions, we skip auto-detection for `copy_start`. Then the requests will use `copy` by default. We will leave service to decide whether to succeed or not. From my perspective, the auto-detection is a better-to-have thing. It shouldn't block creating...

Reply:
OK, got it.
Since the automatic detection is better to have, and if there are other types of exceptions thrown, there will be the risk of blocking creating snapshots. I think we can expand to other exceptions to skip automatic detection.
So this way can be consistent with the original logic to the greatest extent, what do you thik of it?

Reply:
modified~